packages/http/src/validator/index.ts

```diff
@@ -41,19 +42,39 @@ const validateOutput: ValidatorFn<IHttpOperation, IHttpResponse> = ({ resource,
 
   return pipe(
     findOperationResponse(resource.responses, element.statusCode),
-    fold<IHttpOperationResponse, IPrismDiagnostic[]>(
-      () => {
-        return [
-          {
-            message: 'Unable to match returned status code with those defined in spec',
-            severity: DiagnosticSeverity.Error,
-          },
-        ];
+    Option.fold<IHttpOperationResponse, IPrismDiagnostic[]>(
+      () => [
+        {
+          message: 'Unable to match returned status code with those defined in spec',
+          severity:
+            element.statusCode >= 200 && element.statusCode <= 299
+              ? DiagnosticSeverity.Error
+              : DiagnosticSeverity.Warning,
+        },
+      ],
+      operationResponse => {
+        const mismatchingMediaTypeError = pipe(
+          Option.fromNullable(operationResponse.contents),
+          Option.map(contents =>
+            pipe(
+              contents,
+              findFirst(c => c.mediaType === mediaType),
+              Option.map<IMediaTypeContent, IPrismDiagnostic[]>(() => []),
+              Option.getOrElse<IPrismDiagnostic[]>(() => [
+                {
+                  message: `The received media type ${mediaType} does not match the one specified in the document`,
```

Comment:
`mediaType` might be undefined which leads to "The received media type **undefined** does not match the one specified in the document". I think undefined mediaType deserves better handling which will improve user experience.

Reply:
that's sadly true https://github.com/stoplightio/prism/pull/650/files#diff-f26acda03be74b077a817c80a629f514R37

Reply:
Whoops, very good catch. I'll check it out and see what we can do here.

Reply:
I've removed the variable printing; I guess we will understand if the error message is good enough or not when we'll have the validation in place with the proxy.